src/index.js

```diff
@@ -416,18 +419,49 @@ export default class AuthnWidget {
   async handleMfaDeviceChange(evt) {
     evt.preventDefault();
     let state = await this.store.getState();
+    if (this.pollPushNoticationState) {
+      clearTimeout(this.pollPushNoticationState);
+    }
     state.status = 'DEVICE_SELECTION_REQUIRED';
     this.render(this.store.getPreviousStore(), state);
   }
 
+  registerMfaUsePasscodeEventHandler() {
+    if (document.getElementById('usePasscode')) {
+      document.getElementById('usePasscode')
+        .addEventListener('click', this.handleMfaUsePasscode);
+    }
+  }
+
+  async handleMfaUsePasscode(evt) {
+    evt.preventDefault();
+    clearTimeout(this.pollPushNoticationState);
+    let storeState = this.store.getStore();
+    storeState.status = 'OTP_REQUIRED';
+    this.render(this.store.getPreviousStore(), storeState);
+  }
+
   async postPushNotificationWait() {
+    let storeState = this.store.getStore();
```

Comment:
Is this supposed to poll the state of the API somewhere? I'm confused on how the status or validation errors would change

Reply:
The API state is polled below, but this block is intended to impose OTP_REQUIRED as the state (with any validation errors) when 'Use Passcode' has been clicked and this dangling callback is invoked. We don't poll here because the widget state and server state are out of sync. I know that's not ideal but it matches the P1 implementation.

Reply:
Ah, getState does that call. Forgot about that :)

Looks good